lib/msf/core/exploit/remote/smb/client/psexec.rb

```diff
@@ -95,6 +95,8 @@ def smb_read_file(smbshare, host, file)
   # @param disconnect [Boolean] Disconnect afterwards
   # @return [Boolean] Whether everything went well
   def psexec(command, disconnect=true)
+    remove_socket(self.sock)
```

Comment:
This is off topic but it fixes a race condition I noticed while testing this where when the session is opened the socket would be closed while psexec is still trying to remove the service for cleanup. Without this in place, the session would be opened but the module would hang for a minute before reporting a timeout error as it was trying to delete the service.

The socket is closed on session creation by:
* [`#abort_sockets`](https://github.com/rapid7/metasploit-framework/blob/09fe84faed5fa055df54fdf858ebd0de750eb34f/lib/msf/core/exploit/remote.rb#L48)
* [`#on_session`](https://github.com/rapid7/metasploit-framework/blob/09fe84faed5fa055df54fdf858ebd0de750eb34f/lib/msf/core/payload.rb#L508)